hack/lib/build/environment.sh

```diff
@@ -183,6 +184,12 @@ function os::build::environment::withsource() {
     tar -cf - "${excluded[@]}" . | docker cp - "${container}:${workingdir}"
   fi
 
+  local tmpdir="$(docker inspect -f '{{ range .Config.Env }}{{ if eq (index (split . "=")  0) "TMPDIR" }}{{ index (split . "=") 1 }}{{ end }}{{end}}' ${container})"
```

Comment:
>Why does TMPDIR have to exist? I'm really confused here.

In this case, simply because I am trying to set it under the existing volume that we create and mount at `$workingdir`

>The original reason TMPDIR in the image is /openshifttmp is because old versions of docker set a 64mb limit on it, which is far too small for tmp output. If that's fixed in docker now, we can simply remove that setting and restore it to the default of /tmp.

It looks like this is no longer the case, but it appears that /tmp exists on the container fs, which would still have the potential of running out of disk space, since the default is 10G.

>If there is still a limit, then we want to mount a reasonable TMPDIR in, in which case we don't care where on the FS is inside the container or what the container thinks the dir is. We just want to create a volume and mount it to /tmp and set TMPDIR to /tmp. No need to do this reading, or to make it fall within _output.

Sounds good, I'll go ahead and fix this up to just create a separate volume and bind mount it to /tmp